esp-hal/src/gpio.rs

```diff
@@ -38,6 +43,10 @@ pub type NoPinType = Gpio0<Unknown>;
 /// Convenience constant for `Option::None` pin
 pub const NO_PIN: Option<NoPinType> = None;
 
+#[cfg(feature = "async")]
+static USER_INTERRUPT_HANDLER: Mutex<RefCell<Option<unsafe extern "C" fn()>>> =
```

Comment:
I think we can avoid the option here by setting the default value to the `DefaultHandler`, right? We can also drop the RefCell and use `Cell` to replace the value to save a byte :D.